### DISCUSSION TITLE
put focus on active q route tab component?

### DISCUSSION THREAD

**Discussion by laferte-tech** [quasarframework/quasar#10034] [General - Components / Directives / etc] [answered]
Hello,

let's say i have 10 q-route-tab component inside a q-tabs component. When the screen is not big enough to display all of them, we can use arrows to navigate but the "focus" is not put on the active q route tab but always on the first one.
For example, from another page, if i navigate to 9nth component in the list the active class is working great but the focus is not put on this component, it's always the beginning (ie the first q route tab) of qtabs that is displayed and the user need to scroll to the end to see that it's the 9nth component that is active.
Not sure if it's clear but i would like to know if it's possible to put focus on the active q route tab component ? (to avoid to have to scroll the whole list to find it).

Thanks a lot

**Comment by hawkeye64** [comment]
You can use `onBeforeUnmount` to save the current tab to a variable outside of your component containing the q-tabs. You can then use `onMounted` to set the v-model used by q-tabs to ensure the correct tab is set.

**Reply by laferte-tech** [reply]
Hi,
thanks for your answer but it's not working with v-model. I think my question was not so clear but taking this modified sample from Quasar:
https://codepen.io/grandours35/pen/wvdJPZY

you can see that the "focus' (sorry i don't know how to call it) is not put on the active tab.
We need to scroll to reach "test2" which is the tab that have the v-model.
I would like to have "test2" visible directly when showing the q-tabs.
Maybe i'm missing something?
Thanks for your help

**Reply by hawkeye64** [reply]
I would say this is a legit issue. If the starting tab is not visible, it should be made visible. I'd expect the same from components like a carousel, scrolling list, etc.

**Comment by pdanpdan** [comment]
Check this: https://github.com/pdanpdan/quasar/releases/tag/quasar-pdan-v1.15.22-beta.4

In package.json replace quasar dependency with:
"quasar": "https://github.com/pdanpdan/quasar#quasar-pdan-v1.15.22-beta.4"

Then do a yarn or npm install and start the app again.

Please report if you find something not working.

**Reply by laferte-tech** [reply]
Hi,
Thank you very much, it's working!
Will it be included in the next official release?

Just 2 remarks:
- If we want to see the full qtab element (except for the last one in the list), i need to remove the arrows otherwise the arrow is overlapping the qtab component. Not a big deal but maybe could it be possible to take the arrow size into consideration?

- For information, I'm using Docker and it's mandatory to add Git to the NodeJs image to be able to run yarn install:
`RUN apk add --no-cache git`

Thanks again

**Reply by pdanpdan** [reply]
Use arrows-outside (if I don't mix the props). Else the arrow is expected to overlap the tab.

**Reply by laferte-tech** [reply]
Yes, you are right sorry. I was using mobile-arrows in the same time and i was seeing this in mobile view.
Thanks a lot!

**Reply by laferte-tech** [reply]
Hi again @pdanpdan,
I know have this warning using Q-editor:

```
[Vue warn]: Failed to resolve directive: key-group-navigation

(found in <QEditor>)
```
I see in your changelog that you have added this directive, do i need to change something on my side or is it a bug?
Thank you

**Reply by pdanpdan** [reply]
What version are you using? It should be 1.15.22-beta.4
I know it was a bug, but it was some 20 releases ago

**Reply by laferte-tech** [reply]
Yes I'm using the version you mentioned:

```
    Dev mode.......... ssr
    Pkg quasar........ v1.15.22-beta.4
    Pkg @quasar/app... v2.2.10
    Transpiled JS..... yes (Babel)
```
and more info on the error:

```
[Vue warn]: Failed to resolve directive: key-group-navigation

(found in <QEditor>) vue.runtime.esm.js:619
    VueJS 21
    refreshToolbar QEditor.js:430
    (Asynchrone : setTimeout handler)
    refreshToolbar QEditor.js:428
    mounted QEditor.js:469
    VueJS 18
    init vue-router.esm.js:2989
    init vue-router.esm.js:2988
    updateRoute vue-router.esm.js:2409
    transitionTo vue-router.esm.js:2263
    confirmTransition vue-router.esm.js:2397
    step vue-router.esm.js:2001
    step vue-router.esm.js:2005
    iterator vue-router.esm.js:2379
```
Just loading q-editor on my page is causing the error

**Reply by pdanpdan** [reply]
Found the problem, please try 1.15.22-beta.5

**Reply by laferte-tech** [reply]
It's fixed! Thanks a lot

**Reply by pdanpdan** [reply]
thanks for catching it

**Reply by laferte-tech** [reply]
Hi @pdanpdan 
I see that the fix is still not in 1.16 (based on your release of 1.16-beta 3 in your repo)
Do you know if someday it will be added to the main branch?

Thanks a lot

**Reply by pdanpdan** [reply]
The fixes in my branch are not expected to land in the Quasar v1, sorry - it's going to be only critical fixes for V1.
But I'll keep updating my branch and keep it in sync with official one. So as long as you stay on v1 you can use it.

**Reply by pdanpdan** [reply]
1.16-beta3 should have the fix

**Reply by laferte-tech** [reply]
Hi, sorry for late answer.
Ok thanks for the information, i understand.
And thanks also for your work

**Comment by rstoenescu** [comment, marked as answer]
Focus enhancement will be available in Quasar v2.4 and v1.17
